### ISSUE TITLE
 [ERRO] [node4] [Driver] [grpc] rpc error: code = Internal desc = mount failed: exit status 32

### ISSUE THREAD

**Issue by gofman8** [Nexenta/edgefs-csi#16]
hi! I am trying to mount pv to pod with help of NFS dynamic provisioning and met problem during mounting process.
Error:
Jan  5 21:13:43.214 [ERRO] [node4] [Driver] [NodePublishVolume()] Failed to mount volume 10.174.65.38:/nfs/pvc-c26498bd-c3c5-4a5a-b946-57b4c1179f55. Error: mount failed: exit status 32
Mounting command: mount
Mounting arguments: -t nfs -o vers=3,tcp 10.x.x.x:/nfs/pvc-c26498bd-c3c5-4a5a-b946-57b4c1179f55 /var/lib/kubelet/pods/ab2eb907-94a5-4dc3-b55c-ebb9bd109d56/volumes/kubernetes.io~csi/pvc-c26498bd-c3c5-4a5a-b946-57b4c1179f55/mount
Output: mount.nfs: rpc.statd is not running but is required for remote locking.
mount.nfs: Either use '-o nolock' to keep locks local, or start statd.
mount.nfs: an incorrect mount option was specified
OS: Centos 7
rpcbind and nfs-utils packages installed.
rpcbind service started

**Comment by gofman8** [comment]
Thanks. I have found root problem. For Centos 7 this services must be running:
systemctl start nfslock
systemctl start rpc-statd.service

**Comment by Aerius42** [comment]
Hi ! I'm trying to do the same but I have this message in the driver:

Error:
Mar 21 02:38:40.992[31m [ERRO] [node1] [Driver] [NodePublishVolume()] [0mFailed to mount volume 10.96.138.144:/nfs-csi-tenant/pvc-66bc0aa7-a2df-41f2-9116-f04c5da73500. Error: mount failed: exit status 32
Mounting command: mount
Mounting arguments: -t nfs -o vers=3,tcp 10.x.x.x:/nfs-csi-tenant/pvc-66bc0aa7-a2df-41f2-9116-f04c5da73500 /var/lib/kubelet/pods/83ba7150-6682-4b40-b379-66c0bce5ff47/volumes/kubernetes.io~csi/pvc-66bc0aa7-a2df-41f2-9116-f04c5da73500/mount
Output: mount.nfs: Operation timed out

Mar 21 02:38:40.992[31m [ERRO] [node1] [Driver] [grpc] [0mrpc error: code = Internal desc = mount failed: exit status 32
Mounting command: mount
Mounting arguments: -t nfs -o vers=3,tcp 10.x.x.x:/nfs-csi-tenant/pvc-66bc0aa7-a2df-41f2-9116-f04c5da73500 /var/lib/kubelet/pods/83ba7150-6682-4b40-b379-66c0bce5ff47/volumes/kubernetes.io~csi/pvc-66bc0aa7-a2df-41f2-9116-f04c5da73500/mount
Output: mount.nfs: Operation timed out

OS: RHEL7

rpcbind and nfs-utils packages installed.
rpcbind service started

I tried your solution with:
systemctl start nfslock
systemctl start rpc-statd.service
But nothing new happened.

**Comment by dyusupov** [comment]
firewall is on?

**Comment by Aerius42** [comment]
Thank you for your answer, it was indeed a networking problem but not the firewall.
Actually, I had to modify the edgefs-nfs-csi-node in edgefs-nfs-csi-driver.yaml to use the k8s network and see the internal services. I set:
hostNetwork: false

Then, I had the same issue as @SkySonR but the rpcbind was already up on the host but not in the container. So, I add this to its driver:
lifecycle:
  postStart:
    exec:
      command: ["/sbin/rpcbind"]

I don't know why rpcbind didn't start at boot inside the container but that's do the thing. =D
